34th-Sopt-Assignment/34th-Sopt-Assignment/BottomSheetViewController.swift

```diff
@@ -0,0 +1,182 @@
+//
+//  BottomSheetViewController.swift
+//  34th-Sopt-Assignment
+//
+//  Created by 김민서 on 4/15/24.
+//
+
+import UIKit
+import Then
+import SnapKit
+
+
+class BottomSheetViewController: UIViewController {
+  
+    
+    private let bottomSheetView = UIView()
+    private let dimmedView = UIView()
+    private let bar = UIView()
+    private let nicknameTextfield = UITextField()
+    private let airpodMax = UILabel()
+    private var bottomSheetViewTopConstraint: NSLayoutConstraint!
+    private lazy var saveButton = UIButton()
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setStyle()
+        setLayout()
+    }
+    
+
+    private func setLayout() {
+        [
+            dimmedView,
+            bottomSheetView,
+            bar,
+            nicknameTextfield,
+            airpodMax,
+            saveButton
+        ].forEach { [weak self] view in
+            guard let self = self else { return }
+            self.view.addSubview(view)
+        }
+        dimmedView.translatesAutoresizingMaskIntoConstraints = false
+        dimmedView.snp.makeConstraints {
+            $0.top.equalToSuperview()
+            $0.leading.equalToSuperview()
+            $0.trailing.equalToSuperview()
+            $0.bottom.equalToSuperview()
+        }
+        bottomSheetView.translatesAutoresizingMaskIntoConstraints = false
+
+        let topConstant = view.frame.height * 0.5
+
+        bottomSheetView.snp.makeConstraints {
+            $0.top.equalToSuperview().offset(topConstant)
+            $0.leading.equalToSuperview()
+            $0.trailing.equalToSuperview()
+            $0.bottom.equalToSuperview()
+        }
+        bar.snp.makeConstraints {
+            $0.bottom.equalTo(bottomSheetView.snp.top).offset(-15)
+            $0.center.equalToSuperview()
+            $0.width.equalTo(56)
+            $0.height.equalTo(7)
+        }
+        nicknameTextfield.snp.makeConstraints {
+            $0.top.equalTo(bottomSheetView.snp.top).inset(98)
+            $0.leading.equalToSuperview().inset(20)
+            $0.trailing.equalToSuperview().inset(20)
+            $0.height.equalTo(52)
+            $0.width.equalTo(56)
```

Comment:
```suggestion
            $0.leading.equalToSuperview().inset(20)
            $0.trailing.equalToSuperview().inset(20)
            $0.height.equalTo(52)
```
leading과 trailing을 잡아주시면 width가 자동으로 잡아지게 됩니다!
해당 코드에서inset값을 더 크게 적고 width를 삭제 하거나 
```suggestion
            $0.height.equalTo(52)
            $0.width.equalTo(56)
            $0.centerX.equalToSuperview()
```
이런 식으로 적어주시면 좋을 것 같습니다!